zebra-state/src/service/read/tree.rs

```diff
@@ -44,26 +48,71 @@ where
         .or_else(|| db.sapling_tree_by_hash_or_height(hash_or_height))
 }
 
-/// Returns the Sapling
-/// [`NoteCommitmentSubtree`] specified by an
-/// index, if it exists in the non-finalized `chain` or finalized `db`.
-#[allow(unused)]
-pub fn sapling_subtree<C>(
+/// Returns a list of Sapling [`NoteCommitmentSubtree`]s starting at `start_index`.
+/// If `limit` is provided, the list is limited to `limit` entries.
+///
+/// If there is no subtree at `start_index` in the non-finalized `chain` or finalized `db`,
+/// the returned list is empty. Otherwise, subtrees are continuous and consistent up to the tip.
```

Comment:
The last sentence doesn't mention if it's the finalized or non-finalized tip. We could move the comment on correctness from below to here.